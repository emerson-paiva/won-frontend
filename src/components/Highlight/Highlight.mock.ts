export default {
  title: "Read Dead it's back",
  subTitle: "Come see John's new adventures",
  backgroundImage: '/img/red-dead-img.jpg',
  buttonLabel: 'Buy now',
  buttonLink: '/rdr2'
}
