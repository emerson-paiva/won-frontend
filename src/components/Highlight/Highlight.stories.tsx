import { Meta, StoryObj } from '@storybook/react'
import mock from './__data__/mocks'
import { Highlight, HighlightProps } from '.'

export default {
  title: 'Highlight',
  component: Highlight,
  args: mock
} as Meta

export const Default: StoryObj<HighlightProps> = {}

export const WithFloatImage: StoryObj<HighlightProps> = {
  args: {
    floatImage: '/img/red-dead-float.png'
  }
}
