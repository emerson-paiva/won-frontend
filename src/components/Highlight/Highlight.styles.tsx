import styled, { css } from 'styled-components'
import media from 'styled-media-query'
import { HighlightProps } from '.'

type WrapperProps = Pick<HighlightProps, 'backgroundImage' | 'alignment'>

const wrapperModifiers = {
  right: () => css`
    grid-template-areas: 'floatimage content';
    grid-template-columns: 1.3fr 2fr;

    ${Content} {
      text-align: right;
    }
  `,
  left: () => css`
    grid-template-areas: 'content floatimage';
    grid-template-columns: 2fr 1.3fr;

    ${Content} {
      text-align: left;
    }

    ${FloatImage} {
      justify-self: end;
    }
  `
}

export const Wrapper = styled.section<WrapperProps>`
  ${({ alignment, backgroundImage }) => css`
    background-image: url(${backgroundImage});
    background-position: center center;
    background-size: cover;
    position: relative;
    height: 23rem;
    display: grid;

    &::after {
      content: '';
      position: absolute;
      height: 100%;
      width: 100%;
      background-color: rgba(0, 0, 0, 0.6);
    }

    ${media.greaterThan('medium')`
      height: 32rem;
    `}

    ${wrapperModifiers[alignment!]()}
  `}
`

export const FloatImage = styled.img`
  ${({ theme }) => css`
    grid-area: floatimage;
    z-index: ${theme.layers.base};
    max-height: 23rem;
    max-width: 100%;
    align-self: end;

    ${media.greaterThan('medium')`
      max-height: 32rem;
    `}
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    grid-area: content;
    padding: ${theme.spacings.xsmall};
    z-index: ${theme.layers.base};

    ${media.greaterThan('medium')`
      align-self: end;
      padding: ${theme.spacings.large};
    `}
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.large};
    font-weight: ${theme.font.bold};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xxlarge};
    `}
  `}
`

export const SubTitle = styled.h3`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.light};
    margin-bottom: ${theme.spacings.medium};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.large};
    `}
  `}
`
