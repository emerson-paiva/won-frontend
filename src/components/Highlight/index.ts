export { default } from './Highlight'
export * from './Highlight'
