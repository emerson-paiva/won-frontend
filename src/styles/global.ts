import {
  createGlobalStyle,
  css,
  DefaultTheme,
  GlobalStyleComponent
} from 'styled-components'

type GlobalStyleProps = {
  removeBg?: boolean
}

const GlobalStyles: GlobalStyleComponent<
  GlobalStyleProps,
  DefaultTheme
> = createGlobalStyle`
  @font-face {
    font-family: 'Poppins';
    font-style: normal;
    font-weight: 300;
    font-display: swap;
    src: local('Poppins Light'), local('Poppins-Light'),
        url('/fonts/poppins-v12-latin-300.woff2') format('woff2');
  }

  @font-face {
    font-family: 'Poppins';
    font-style: normal;
    font-weight: 400;
    font-display: swap;
    src: local('Poppins Regular'), local('Poppins-Regular'),
        url('/fonts/poppins-v12-latin-regular.woff2') format('woff2');
  }

  @font-face {
    font-family: 'Poppins';
    font-style: normal;
    font-weight: 500;
    font-display: swap;
    src: local('Poppins Medium'), local('Poppins-Medium'),
        url('/fonts/poppins-v12-latin-500.woff2') format('woff2');
  }

  *,
  *::after,
  *::before {
    margin: 0;
    padding: 0;
    box-sizing: inherit;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  ${({ theme, removeBg }) => css`
    html {
      box-sizing: border-box;
      font-size: 62.5%;
    }

    body {
      font-family: ${theme.font.family};
      font-size: ${theme.font.sizes.medium};

      ${!removeBg &&
      css`
        background-color: ${theme.colors.mainBg};
      `}
    }
  `}

`

export default GlobalStyles
