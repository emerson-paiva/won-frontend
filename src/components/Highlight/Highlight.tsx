import Button from 'components/Button'
import * as S from './Highlight.styles'

export type HighlightProps = {
  title: string
  subTitle: string
  backgroundImage: string
  buttonLabel: string
  buttonLink: string
  backgroundImage: string
}

const Highlight = ({
  title,
  subTitle,
  backgroundImage,
  buttonLabel,
  buttonLink
}: HighlightProps) => (
  <S.Wrapper backgroundImage={backgroundImage}>
    <S.Content>
      <S.Title>{title}</S.Title>
      <S.SubTitle>{subTitle}</S.SubTitle>
      <Button as="a" href={buttonLink}>
        {buttonLabel}
      </Button>
    </S.Content>
  </S.Wrapper>
)

export default Highlight
