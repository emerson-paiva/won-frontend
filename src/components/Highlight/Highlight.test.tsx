import { renderWithTheme } from '@/utils/tests/helpers'
import { screen } from '@testing-library/react'

import { Highlight } from '.'

const props = {
  title: 'Heading 1',
  subtitle: 'Heading 2',
  backgroundImage: '/img/red-dead-img.jpg',
  buttonLabel: 'Buy now',
  buttonLink: '/rdr2'
}

import * as S from './Highlight.styles'

describe('<Highlight />', () => {
  it('should render headings and button', () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(
      screen.getByRole('heading', { name: /heading 1/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /heading 2/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('link', { name: /buy now/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it.skip('should render background image', () => {
    renderWithTheme(<Highlight {...props} />)

    expect(
      screen.getByRole('img', { name: `${props.title} background` })
    ).toHaveAttribute('src', `${props.backgroundImage}`)
  })

  it.skip('should render float image', () => {
    renderWithTheme(<Highlight {...props} floatImage="/float-image.png" />)

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      /.+float-image\.png.+/
    )
  })

  it('should render align right by default', () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(container.firstChild).toHaveStyleRule(
      'grid-template-areas',
      "'floatimage content'"
    )

    expect(container.firstChild).toHaveStyleRule('text-align', 'right', {
      modifier: `${S.Content}`
    })
  })

  it('should render align left when argument is passed', () => {
    const { container } = renderWithTheme(
      <Highlight {...props} alignment="left" />
    )

    expect(container.firstChild).toHaveStyleRule(
      'grid-template-areas',
      "'content floatimage'"
    )

    expect(container.firstChild).toHaveStyleRule('text-align', 'left', {
      modifier: `${S.Content}`
    })
  })
})
