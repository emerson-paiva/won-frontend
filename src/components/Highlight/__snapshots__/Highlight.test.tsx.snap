// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should render headings and button 1`] = `
.c5 {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(180deg, #ff5f5f 0%, #f062c0 50%);
  color: #FAFAFA;
  cursor: pointer;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  text-decoration: none;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c5:hover {
  background: linear-gradient(180deg, #e35565 0%, #d958a6 50%);
}

.c0 {
  position: relative;
  height: 23rem;
  display: grid;
  grid-template-areas: 'floatimage content';
  grid-template-columns: 1.3fr 2fr;
}

.c0::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.6);
}

.c0 img {
  position: absolute;
  object-fit: cover;
}

.c0 .c1 {
  text-align: right;
}

.c2 {
  grid-area: content;
  z-index: 10;
  padding: 1.6rem;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c4 {
  font-size: 1.4rem;
  font-weight: 300;
  color: #FAFAFA;
  margin-bottom: 3.2rem;
}

@media (min-width: 768px) {
  .c0 {
    height: 32rem;
  }
}

@media (min-width: 768px) {
  .c2 {
    align-self: end;
    padding: 4.0rem;
  }
}

@media (min-width: 768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width: 768px) {
  .c4 {
    font-size: 1.8rem;
  }
}

<section
  alignment="right"
  class="c0"
  data-cy="highlight"
>
  <img
    alt="Heading 1 background"
    data-nimg="fill"
    decoding="async"
    loading="lazy"
    sizes="100vw"
    src="/_next/image?url=%2Fimg%2Fred-dead-img.jpg&w=3840&q=75"
    srcset="/_next/image?url=%2Fimg%2Fred-dead-img.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg%2Fred-dead-img.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg%2Fred-dead-img.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg%2Fred-dead-img.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg%2Fred-dead-img.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg%2Fred-dead-img.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg%2Fred-dead-img.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg%2Fred-dead-img.jpg&w=3840&q=75 3840w"
    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
  />
  <div
    class="c1 c2"
  >
    <h2
      class="c3"
    >
      Heading 1
    </h2>
    <h3
      class="c4"
    >
      Heading 2
    </h3>
    <a
      class="c5"
      href="/rdr2"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</section>
`;
